modulemd/include/private/modulemd-yaml.h

```diff
@@ -79,33 +120,108 @@ G_DEFINE_AUTO_CLEANUP_CLEAR_FUNC (yaml_parser_t, yaml_parser_delete);
   g_auto (yaml_parser_t) _parser;                                             \
   yaml_parser_initialize (&_parser)
 
+/**
+ * MMD_INIT_YAML_EMITTER:
+ * @_emitter: A variable name to use for the new emitter object.
+ *
+ * This convenience macro allocates and initializes a new libyaml emitter
+ * object named @_emitter.
+ *
+ * Since: 2.0
+ */
 #define MMD_INIT_YAML_EMITTER(_emitter)                                       \
   g_auto (yaml_emitter_t) _emitter;                                           \
   yaml_emitter_initialize (&_emitter)
 
+/**
+ * MMD_INIT_YAML_EVENT:
+ * @_event: A variable name to use for the new event object.
+ *
+ * This convenience macro allocates and initializes a new libyaml event object
+ * named @_event.
+ *
+ * Since: 2.0
+ */
 #define MMD_INIT_YAML_EVENT(_event)                                           \
   g_auto (yaml_event_t) _event;                                               \
   memset (&(_event), 0, sizeof (yaml_event_t))
 
+/**
+ * MMD_INIT_YAML_STRING:
+ * @_emitter: A libyaml emitter object.
+ * @_string: A variable name to use for the new yaml string object.
+ *
+ * This convenience macro allocates and initializes a new yaml string object
+ * named @_string and associates it as the output target for the libyaml
+ * emitter object @_emitter.
+ *
+ * Since: 2.0
+ */
 #define MMD_INIT_YAML_STRING(_emitter, _string)                               \
   g_autoptr (modulemd_yaml_string) yaml_string =                              \
     g_malloc0_n (1, sizeof (modulemd_yaml_string));                           \
   yaml_emitter_set_output (_emitter, write_yaml_string, (void *)yaml_string)
 
+/**
+ * MMD_REINIT_YAML_STRING:
+ * @_emitter: A libyaml emitter object to reinitialize.
+ * @_string: A variable name to reinitialize with a new yaml string object.
+ *
+ * This convenience macro deletes then initializes a new libyaml emitter named
+ * @_emitter, then deletes, reallocates, and initializes a new yaml string
+ * object named @_string and associates it as the output target for @_emitter.
+ *
+ * Since: 2.0
+ */
 #define MMD_REINIT_YAML_STRING(_emitter, _string)                             \
   yaml_emitter_delete (_emitter);                                             \
   yaml_emitter_initialize (_emitter);                                         \
   g_clear_pointer (&_string, modulemd_yaml_string_free);                      \
   yaml_string = g_malloc0_n (1, sizeof (modulemd_yaml_string));               \
   yaml_emitter_set_output (_emitter, write_yaml_string, (void *)yaml_string)
 
+/**
+ * YAML_PARSER_PARSE_WITH_EXIT_FULL:
+ * @_parser: A libyaml parser object positioned at the beginning of an event.
+ * @_returnval: The value to return in case of of a parsing error.
+ * @_event: Returns the type of libyaml event that was parsed.
+ * @_error: A #GError that will return the reason for a parsing or validation
+ * error.
+ *
+ * DIRECT USE OF THIS FUNCTION SHOULD BE AVOIDED. This is the internal
```

Comment:
```suggestion
 * DIRECT USE OF THIS MACRO SHOULD BE AVOIDED. This is the internal
```
This just jumped out at me right now.